src/main/java/pl/multiplex/multiplexreservation/manager/ReservationManager.java

```diff
@@ -1,4 +1,163 @@
+        return screeningRepo.findScreeningByTimeAndDate(date,time);
+    }
+    public List<String> showByTimeAndMinMaxDayOnlyTitle(int date, int timeMin, int timeMax) {
+        List<Screening> availableScreenings =  screeningRepo.findScreeningByMinMaxTimeAndDate(date, timeMin, timeMax);
+        Set<String> availableMovies = new HashSet<String>();
+        List<String> moviesList = new LinkedList<String>();
+        Screening scr;
+        Iterator screenIter = availableScreenings.iterator();
+        while(screenIter.hasNext()){
+            scr = (Screening) screenIter.next();
+            if(!availableMovies.contains(scr.getMovieId())){
+                availableMovies.add(scr.getMovieId().getTitle());
+            }
+        }
+
+        if(!availableMovies.isEmpty()){
+            Iterator<String> moviesIter = availableMovies.iterator();
+            while(moviesIter.hasNext()){
+                moviesList.add(moviesIter.next());
+            }
+        }
+
+        Collections.sort(moviesList);
+        return moviesList;
+    }
+
+    public List<Screening> showByTimeAndMinMaxDay(int date, int timeMin, int timeMax) {
+        List<Screening> availableScreenings =  screeningRepo.findScreeningByMinMaxTimeAndDate(date, timeMin, timeMax);
+        return availableScreenings;
+    }
+
+    public List<Seat> getSeatsByScreeningId(Long screeningId){
+        List<Seat>seats = new LinkedList<Seat>();
+        Screening scr = screeningRepo.findScreeningById(screeningId);
+        if(!(scr == null)){
+            Set<Seat> seatsSet = scr.getCinemaRoomId().getSeats();
+            Iterator<Seat> seatIter = seatsSet.iterator();
+            while(seatIter.hasNext()){
+                seats.add(seatIter.next());
+            }
+            List<Reservation> reservedSeats = reservationRepo.getReservedSeats(screeningId);
+            Iterator<Reservation> reservationsIter = reservedSeats.iterator();
+            while(reservationsIter.hasNext()){
+                Seat reservedSeat = reservationsIter.next().getSeatId();
+                if(seats.contains(reservedSeat)){
+                    seats.remove(reservedSeat);
+                }
+            }
+        }
+        return seats;
+    }
+
+    public void postReservation(Long userId, Long screeningId, Long seatId, String ticketType){
+        User user = userRepo.findUserById(userId);
+        Screening scr = screeningRepo.findScreeningById(screeningId);
+        Seat seat = scr.getCinemaRoomId().getSeat(seatId);
+        double price = 0;
+        boolean correct = false;
+        if(ticketType.equals("adult")){
+            price = 25;
+            correct = true;
+        }
+        else if(ticketType.equals("student")){
+            price = 18;
+            correct = true;
+        }
+        else if(ticketType.equals("child")){
+            price = 12.5;
+            correct = true;
+        }
+        else {System.out.println("Podaj wlasciwy typ biletu");}
```

Comment:
Jak to się ma do REST?

Reply:
Myślę, że można by było zrobić w REST Controllerze funkcje, która by zwracała Stringa z komunikatem i potem w innych funkcjach korzystać z niej, ale nie wiem czy to ma sens.